app/models/concerns/authorizable.rb

```diff
@@ -6,6 +6,33 @@ def authorized?(permission)
       return false if User.current.nil?
       User.current.can?(permission, self)
     end
+
+    def permission_name(name)
+      Permission.where(:resource_type => Permission.resource_name(self.class)).where(["#{Permission.table_name}.name LIKE ?", "#{name}_%"]).first.try(:name)
+    end
+
+    def check_permissions_after_save
+      authorizer = Authorizer.new(User.current)
+      creation = self.id_changed?
+      name = permission_name(creation ? :create : :update)
+
+      unless authorizer.can?(name, self)
+        errors.add :base, "You don't have permission #{name} with attributes that you have specified"
+        if creation
```

Comment:
I think it's good to put the comment not only to the PR, but actually into the code for future generations